### PR TITLE
Better validation of notification orders

### PR DESCRIPTION
**Pull request by martinothamar**
## Description
Implements validation of more order fields based on constraints from downstream providers. I came across this since I was wondering what the input constraints was when developing the SMS client in apps

## Related Issue(s)
N/A

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Comment by martinothamar** [comment]
@acn-sbuad since this would be a breaking change I'm not sure if this is something you want. But I think a change like this is good because
* Users get bad request errors early (as opposed to only noticing when reading the status, I'm assuming errors due to service limits would appear somewhere in the status/delivery reports).
* Unbounded input is not a good idea in terms of security, upstream applications could affect service availability and performance (either due to bugs or security issues) I think

If we do something like this, we should document the limits as well

File: src/Altinn.Notifications/Validators/EmailNotificationOrderRequestValidator.cs [hunk 1]
```diff
@@ -17,8 +17,8 @@ public class EmailNotificationOrderRequestValidator : AbstractValidator<EmailNot
     public EmailNotificationOrderRequestValidator()
     {
         RuleFor(order => order.Recipients)
-            .NotEmpty()
-            .WithMessage("One or more recipient is required.")
+            .Must(r => r?.Count is >= 1 and <= 50) // Azure Communication Services has max 10 recipients
```
Comment (acn-sbuad) [inline src/Altinn.Notifications/Validators/EmailNotificationOrderRequestValidator.cs:20]: Number of recipients for an order doesn't directly translate to how many recipients are set in the to field of an email. One email is sent per recipient.

Comment (martinothamar) [reply src/Altinn.Notifications/Validators/EmailNotificationOrderRequestValidator.cs:20]: Ahh, makes sense! Should there be still some upper bound? Like should [Array.MaxLength (2,147,483,591)](https://learn.microsoft.com/en-us/dotnet/api/system.array.maxlength?view=net-8.0) number of recipients be allowed?

Comment (acn-sbuad) [reply src/Altinn.Notifications/Validators/EmailNotificationOrderRequestValidator.cs:20]: I would like to keep it open for now. We are still working on analyzing how batch orders should be handled and will most likely have a separate endpoint for posting extra large orders (think information about tax forms being available), once this has been specified we will be able to set a reasonable limit for this endpoint. Current use cases, e.g. notifications from an app triggered by the user or by instantiation will not generate any significant number of recipients is our current hypothesis.

**Pull request closed without merging**